OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -168,7 +168,8 @@ void Attack(Actor self, Actor targetActor, bool allowMove)
 		Actor ChooseTarget(Actor self, WDist range, bool allowMove)
 		{
 			var inRange = self.World.FindActorsInCircle(self.CenterPosition, range)
-				.Where(a => !a.Info.HasTraitInfo<AutoTargetIgnoreInfo>());
+				.Where(a =>
+				!a.TraitsImplementing<IPreventsAutoTarget>().Any(t => t.PreventAutoTarget(a, self)));
```

Comment:
Our style convention is to indent cases like this 2nd line with another tab (to make it more visible that the line belongs to the `Where` above).